app/screens/custom_status/custom_status_modal.tsx

```diff
@@ -86,17 +86,30 @@ class CustomStatusModal extends PureComponent<Props, State> {
         };
     }
 
+    componentDidMount() {
+        Navigation.events().bindComponent(this);
+    }
+
     navigationButtonPressed() {
         this.handleSetStatus();
     }
 
     handleSetStatus = () => {
         const {emoji, text} = this.state;
-        const customStatus = {
-            emoji: emoji || 'speech_balloon',
-            text: text.trim(),
-        };
-        this.props.actions.setCustomStatus(customStatus);
+        const isStatusSet = emoji || text;
+        if (isStatusSet) {
+            const {customStatus} = this.props;
+            const isStatusSame = customStatus.emoji === emoji && customStatus.text === text;
+            if (!isStatusSame) {
+                const status = {
+                    emoji: emoji || 'speech_balloon',
+                    text: text.trim(),
+                };
+                this.props.actions.setCustomStatus(status);
+            }
+        } else {
+            this.props.actions.unsetCustomStatus();
```

Comment:
Don't make this API call if the status was not set initially.
Maybe you can disable the `Done` button in this case.